co/component.py

```diff
@@ -174,7 +173,10 @@ def _translate_feature(feature, component, tt=None):
         # TODO create a "broken reference" object and re-attach here
         return feature._shift(offset, component)
 
-    def mutate(self, mutations, strict=True):
+    def _reallocate(self, feature, start, stop):
```

Comment:
Also, move the logic from feature._move in here. Or you can make it a class method on Feature, but then you don't need this function here.